dataproc_jupyter_plugin/services/dataproc.py

```diff
@@ -52,27 +58,146 @@ async def list_clusters(self, page_size, page_token):
                     return resp
                 else:
                     return {
-                        "error": f"Failed to fetch clusters: {response.status} {await response.text()}"
+                        "error": f"Failed to fetch runtimes: {response.status} {await response.text()}"
                     }
+        except Exception as e:
+            self.log.exception(f"Error fetching runtime list: {str(e)}")
+            return {"error": str(e)}
+
+    async def list_clusters(self, page_size, page_token):
+        try:
+            # Create a client
+            client = dataproc.ClusterControllerAsyncClient(
+                client_options={
+                    "api_endpoint": f"us-central1-dataproc.googleapis.com:443"
+                },
+                credentials=oauth2.Credentials(self._access_token),
+            )
+
+            # Initialize request argument(s)
+            request = dataproc.ListClustersRequest(
+                project_id=self.project_id,
+                page_size=int(page_size),
+                page_token=page_token,
+                region=self.region_id,
+            )
+
+            # Make the request
+            page_result = await client.list_clusters(request=request)
+            clusters_list = []
 
+            # Handle the response
+            async for response in page_result:
+                clusters_list.append(json.loads(proto.Message.to_json(response)))
+
+            return clusters_list
         except Exception as e:
-            self.log.exception("Error fetching cluster list")
+            self.log.exception(f"Error fetching cluster list")
```

Comment:
Everywhere in this file where we log an exception, include the actual exception in the log message.